beet/src/struct.ts

```diff
@@ -104,6 +104,25 @@ export class BeetStruct<Class, Args = Partial<Class>>
     return [writer.buffer, writer.offset]
   }
 
+  /**
+   * Get the offset of a structure field.
+   *
+   * @param fieldName the structure field name of interest.
+   * @returns the offset in bytes to the start of `fieldName`
+   * within the structure. Throw an error if `fieldName` is not a field
+   * within the structure.
+   */
+  offsetOf(fieldName: string): number  {
+    let offset = 0;
+    for (const fd of this.fields) {
+      if (fd[0] === fieldName) {
+        return offset;
+      }
+      offset += fd[1].byteSize;
```

Comment:
I think this is problematic as not all fields have a fixed length and thus you'd need a value or buffer in the case that you want to determine the field offset if there is a field before the field you're trying to get the offset for.

For more info have a look around [here](https://github.com/metaplex-foundation/beet/blob/master/beet-solana/src/gpa/index.ts#L137)